soroban-sdk/src/logs.rs

```diff
@@ -125,6 +125,18 @@ impl Logs {
         if cfg!(debug_assertions) {
             let env = self.env();
             env.log_from_slice(msg, args).unwrap();
+
+            #[cfg(any(test, feature = "testutils"))]
+            {
+                use crate::testutils::Logs;
+                self.env().with_debug_mode(|| {
```

Comment:
Why do you need to wrap this in `with_debug_mode`? 
If I understood correctly, you want the `self.all().last()` call (which externalizes the event) to be budgeted? If so, it should have already been wrapped in debug budget in the host automatically, since that is a debug event.

Reply:
If any contract events (not diagnostice events) were emitted, then the shadow budget wouldn't be used [here](https://github.com/stellar/rs-soroban-env/blob/b00c1249f1e9df027e8ee4d7523c114c7b8ab1e3/soroban-env-host/src/events/internal.rs#L237) right? `externalizes` loops over all emitted events.

Reply:
Oh I see. Then it should probably avoid externalizing all the events, every time it logs a debug event.

Reply:
One way is to add a `get_last()` function inside the host gated by testutils to retrieve and externalize the last event.

Reply:
Yeah I was trying to do this without adding testutils methods specific to events, but that might be cleaner.

Reply:
Ok I updated env to just return the diagnostic events. We still need to filter for the log topic so this was simpler that putting a `last` method in env.